hal/common/critical.c

```diff
@@ -270,7 +270,7 @@ uint32_t core_util_atomic_incr_u32(uint32_t *valuePtr, uint32_t delta)
 }
 
 void *core_util_atomic_incr_ptr(void **valuePtr, ptrdiff_t delta) {
-    return core_util_atomic_incr((uintptr_t)valuePtr, (uintptr_t)delta);
+    return core_util_atomic_incr_u32((uintptr_t)valuePtr, (uintptr_t)delta);
```

Comment:
return a pointer to a stack, that is not valid anymore? Why is return ``void*`` ?

The docs says as I recall ``core_util_atomic_incr_ptr`` operates on bytes, and its u32?